src/en/dynasty/src/eu/kanade/tachiyomi/extension/en/dynasty/Dynasty.kt

```diff
@@ -0,0 +1,698 @@
+        }?.id
+    }
+
+    private fun searchMangaParse(response: Response, filters: FilterList): MangasPage {
+        val typeFilter = filters.firstInstance<TypeFilter>()
+        val includedSeries = typeFilter.checked.contains("Series")
+        val includedChapters = typeFilter.checked.contains("Chapter")
+        val includedDoujins = typeFilter.checked.contains("Doujin")
+
+        val document = response.asJsoup()
+        val entries = LinkedHashSet<MangaEntry>()
+
+        // saves the first entry found
+        // returned if everything was filtered out to avoid "No Results found" error
+        var firstEntry: MangaEntry? = null
+
+        document.select(
+            ".chapter-list a.name[href~=/(series|anthologies|chapters|doujins|issues)/], " +
+                ".chapter-list .doujin_tags a[href~=/doujins/]",
+        ).forEach { element ->
+            var (directory, permalink) = element.absUrl("href")
+                .toHttpUrl().pathSegments
+                .let { it[0] to it[1] }
+            var title = element.ownText()
+
+            if (directory == "chapters") {
+                val seriesPermalink = CHAPTER_SLUG_REGEX.find(permalink)?.groupValues?.get(1)
+
+                if (seriesPermalink != null) {
+                    directory = "series"
+                    permalink = seriesPermalink
+                    title = seriesPermalink.permalinkToTitle()
+                }
+            }
+
+            val entry = MangaEntry(
+                url = "/$directory/$permalink",
+                title = title,
+                cover = getCoverUrl(directory, permalink),
+            )
+
+            // since we convert chapters to their series counterpart, and select doujins from chapters
+            // it is possible to get a certain type even if it is unselected from filters
+            // so don't include in that case
+            if ((!includedSeries && directory == "series") ||
+                (!includedChapters && directory == "chapters") ||
+                (!includedDoujins && directory == "doujins")
+            ) {
+                return@forEach
+            }
+
+            if (firstEntry == null) {
+                firstEntry = entry
+            }
+
+            entries.add(entry)
+        }
+
+        // avoid "No Results found" error in case everything was filtered out from above check
+        if (entries.isEmpty() && firstEntry != null) {
+            entries.add(firstEntry!!)
+        }
+
+        val hasNextPage = document.selectFirst("div.pagination > ul > li.active + li > a") != null
+
+        return MangasPage(
+            mangas = entries.map(MangaEntry::toSManga),
+            hasNextPage = hasNextPage,
+        )
+    }
+
+    override fun getMangaUrl(manga: SManga): String {
+        return baseUrl + manga.url
+    }
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val url = manga.url
+            .removeSuffix("/") // in case of old url having it
```

Comment:
Won't work if URL contains fragment. If it doesn't you can just resolve this comment.